src/som/compiler/Parser.java

```diff
@@ -1482,7 +1518,10 @@ private ExpressionNode nestedBlock(final MethodBuilder builder) throws ProgramDe
       blockPattern(builder);
     }
 
-    String outerMethodName = stripColons(builder.getOuterBuilder().
+    if (builder.getFirstOuterMethod() == null) {
+      throw new java.lang.NullPointerException("The outer context of a block should never be null");
```

Comment:
perhaps this can be turned into an assertion?
```Java
assert builder.getFirstOuterMethod() == null: "The outer context of a block should never be null";
```